src/sentry/static/sentry/app/components/numberDragControl.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import {IconArrow} from 'app/icons';
+import space from 'app/styles/space';
+
+type NumberDragControlProps = {
+  onChange: (delta: number, event: React.MouseEvent<HTMLDivElement>) => void;
+  axis?: 'x' | 'y';
+};
+
+type Props = Omit<React.HTMLAttributes<HTMLDivElement>, keyof NumberDragControlProps> &
+  NumberDragControlProps;
+
+type State = {
+  isClicked: boolean;
+};
+
+class NumberDragControl extends React.Component<Props, State> {
+  state: State = {
+    isClicked: false,
+  };
+
+  render() {
+    const {onChange, axis, ...props} = this.props;
+    const isX = (axis ?? 'x') === 'x';
+
+    return (
+      <Wrapper
+        {...props}
+        onMouseDown={(event: React.MouseEvent<HTMLDivElement>) => {
+          if (event.button !== 0) {
+            return;
+          }
+          event.currentTarget.requestPointerLock();
+          this.setState({isClicked: true});
+        }}
+        onMouseUp={() => {
+          document.exitPointerLock();
+          this.setState({isClicked: false});
+        }}
+        onMouseMove={(event: React.MouseEvent<HTMLDivElement>) =>
```

Comment:
I am in favor of having an inline function for simple things, but maybe you should consider creating an external function, because:

> "Since functions are objects in JavaScript ({} !== {}), the inline function will always fail the prop diff when React does a diff check. Also, an arrow function will create a new instance of the function on each render if it's used in a JSX property. This might create a lot of work for the garbage collector."

Reply:
I'm not too worried, since this won't re-render much